resources/fetchersManager/data_test.go

```diff
@@ -136,8 +136,10 @@ func (s *DataTestSuite) TestDataRun() {
 	d, err := NewData(s.log, interval, timeout, s.registry)
 	s.NoError(err)
 
-	stop := d.Run(s.ctx)
-	defer stop(context.Background(), time.Second)
+	ctx, cancel := context.WithCancel(s.ctx)
+	defer cancel()
```

Comment:
All these `defer cancel()` should be replaced with `defer d.Stop()` except of a specific test that is focused on the context cancellation.